### PR TITLE
SDK Upgrade

### PR DESCRIPTION
**Pull request by danielmarbach**
Support for

- Microsoft.Azure.Cosmos.Table
- Azure.Storage.Blobs

This is the first step in support the Cosmos Table API. In order to make sure when secondary indices are used those entries are compatible with Cosmos DB the partition key value is also propagated to the row key. For people migrating from Azure Storage using 2.4 or earlier they need to make sure the data imported will be using the partition key as a row key so that the document id underneath is properly assigned (due to Cosmos DB table API assuming row key = doc id)

### CODE REVIEW

**Comment by danielmarbach** [comment]
Smoke tested by running our ASP sample, start a few hundred sagas with timeouts, shut down the server, then upgrade to the version with the new SDK, and run the sample again. Everything worked smoothly without any issues.

File: src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs [hunk 1]
```diff
@@ -338,22 +335,21 @@ async Task<TimeoutDataEntity> GetTimeoutData(CloudTable timeoutDataTable, string
 
         async Task SaveCurrentTimeoutState(byte[] state, string stateAddress)
         {
-            var container = cloudBlobClient.GetContainerReference(timeoutStateContainerName);
-            var blob = container.GetBlockBlobReference(stateAddress);
+            var blobClient = cloudBlobClient.GetBlobClient(stateAddress);
+            // TODO: Make this more memory efficient
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs:339]: Memory pool?

File: src/Tests/Sagas/When_storing_saga_with_non_primitive_values.cs [hunk 2]
```diff
@@ -38,13 +37,13 @@ public async Task Should_persist_json_serializable_value()
...
             // assert structure
-            var entity = await GetEntity(saga.Id);
+            var entity = GetEntity(saga.Id);
             var nullableDoubleProp = entity[nameof(NonPrimitiveSerializableSagaData.NullableDouble)];
             Assert.AreEqual(EdmType.Double, nullableDoubleProp.PropertyType);
             Assert.AreEqual(nullableDouble, nullableDoubleProp.DoubleValue);
         }
 
-        static async Task<DictionaryTableEntity> GetEntity(Guid sagaId)
+        static DictionaryTableEntity GetEntity(Guid sagaId)
```
Comment (SeanFeldman) [inline src/Tests/Sagas/When_storing_saga_with_non_primitive_values.cs:46]: 👍

File: src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs (hunk 1 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs:339]: Well probably RecycleableMemoryStream. But this is a topic for another PR. I just noted a TODO for now. I found quite a few other areas that could benefit an overhaul as well which I haven't noted yet down

File: src/Tests/Sagas/When_storing_saga_with_non_primitive_values.cs [hunk 3]
```diff
@@ -54,7 +53,7 @@ static async Task<DictionaryTableEntity> GetEntity(Guid sagaId)
 
             try
             {
-                var tableEntity = (await table.ExecuteQueryAsync(query).ConfigureAwait(false)).SafeFirstOrDefault();
+                var tableEntity = table.ExecuteQuery(query).SafeFirstOrDefault();
```
Comment (SeanFeldman) [inline src/Tests/Sagas/When_storing_saga_with_non_primitive_values.cs:56]: There's no async equivalent?

Comment (danielmarbach) [reply src/Tests/Sagas/When_storing_saga_with_non_primitive_values.cs:56]: Only the query segment version. I can easily switch to that but it felt a bit overkill for the tests to me

Comment (SeanFeldman) [reply src/Tests/Sagas/When_storing_saga_with_non_primitive_values.cs:56]: 👍

**Review by SeanFeldman — APPROVED** [review]

**Comment by SeanFeldman** [comment]
> This is the first step in support the Cosmos Table API. In order to make sure when secondary indices are used those entries are compatible with Cosmos DB the partition key value is also propagated to the row key. For people migrating from Azure Storage using 2.4 or earlier they need to make sure the data imported will be using the partition key as a row key so that the document id underneath is properly assigned (due to Cosmos DB table API assuming row key = doc id)

This is a good segway into creating a migration document.

**Pull request merged**